llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -517,12 +517,21 @@ template <IndexedVersion Version> class RecordWriterTrait {
   // we must use a default constructor with no params for the writer trait so we
   // have a public member which must be initialized by the user.
   MemProfSchema *Schema = nullptr;
+  // The MemProf version to use for the serialization.  The initialization here
+  // is not used.
+  IndexedVersion Version = static_cast<IndexedVersion>(MinimumSupportedVersion);
 
-  RecordWriterTrait() = default;
+  RecordWriterTrait() {
```

Comment:
Can you just mark this constructor deleted via "RecordWriterTrait() = delete;" ?

Reply:
Sure.  I thought deleting it would cause a link error or something, but apparently that's not the case.